longevity_lwt_test.py

```diff
@@ -211,33 +212,13 @@ def start_nemesis(self):
         self.db_cluster.termination_event.clear()
         self.db_cluster.start_nemesis()
 
-    def stop_nemesis_and_repair_cluster(self):
-        # Stop nemesis. Repair on all nodes will be run before data validation, so all nodes should be up
-        # and also prevent case to run repair from nemesis in parallel
-        if self.db_cluster.nemesis_threads:
-            self.db_cluster.stop_nemesis()
-
-        # TODO: Temporary print. Will be removed later
-        self.log.debug('Get rows count in {} MV after sleep and before repair'.format(self.mv_for_not_updated_data))
-        self.get_rows_count(self.db_cluster.nodes[0],
-                            keyspace_name='cqlstress_lwt_example',
-                            table_name='blogposts_not_updated_lwt_indicator')
-
-        for node in self.db_cluster.nodes:
-            self.log.debug('Start nodetool repair on {} node'.format(node.name))
-            node.run_nodetool("repair")
-
-        # TODO: Temporary print. Will be removed later
-        self.log.debug('Get rows count in {} MV after repair'.format(self.mv_for_not_updated_data))
-        self.get_rows_count(self.db_cluster.nodes[0],
-                            keyspace_name='cqlstress_lwt_example',
-                            table_name='blogposts_not_updated_lwt_indicator')
-
     def test_lwt_longevity(self):
         self.test_custom_time()
 
-        # Run repair on all nodes. We need it to fix the case, when part of data wasn't inserted because of timeouts
-        self.stop_nemesis_and_repair_cluster()
+        # Stop nemesis. Prefer all nodes will be run before data validation
+        if self.db_cluster.nemesis_threads:
+            self.db_cluster.stop_nemesis()
```

Comment:
When you stop_nemesis does it wait for the nemesis thread to stop and the actual nemesis that runs to finish?

Reply:
Yes, it joins the nemesis thread (with timeout).